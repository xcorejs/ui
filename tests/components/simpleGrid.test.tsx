import 'jest-styled-components';

import { Box, SimpleGrid, Cell } from '@xcorejs/ui';
import React from 'react';
import renderer from 'react-test-renderer';

test('SimpleGrid component', () => {
  const component = renderer.create(
    <>
      <SimpleGrid display="inline-grid" columns={2} gap="6rem 2rem" mt="3rem" justifyItems="center">
        <Box>Products</Box>
        <Box>Company</Box>
        <Box>Sitemap</Box>
        <Box>Contacts</Box>
        <Box>Disclaimer</Box>
        <Box>Privacy</Box>
      </SimpleGrid>
      <SimpleGrid columns={{ _: 12, sm: 12 }} gap="3rem">
        <Cell gridColumn={{ sm: '1/8' }}>
          AAAAAAA
        </Cell>

        <Cell column={{ sm: '9/13' }} display={{ _: 'none', sm: 'block' }}>
          BBBB
        </Cell>
      </SimpleGrid>
      <SimpleGrid columns={{ _: 1, sm: 12 }} gap="3rem" mb={{ _: '3rem', sm: '8rem' }}>
        <Cell column={{ sm: '1/6' }}>
          AAAAAA
        </Cell>
        <Cell column={{ sm: '6/13' }}>
          BBBBBB
        </Cell>
      </SimpleGrid>
    </>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
