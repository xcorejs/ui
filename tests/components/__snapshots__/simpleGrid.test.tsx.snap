// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleGrid component 1`] = `
Array [
  .c0 {
  display: inline-grid;
  margin-top: 3rem;
  justify-items: center;
  display: grid;
  display: -ms-grid;
  grid-template-columns: repeat(2,1fr);
  grid-gap: 6rem 2rem;
  justify-items: center;
  -ms-grid-columns: 1fr 6rem 1fr;
}

-ms-grid-rows:1fr 2rem 1fr 2rem 1fr .c0 > *:nth-child(1) {
  -ms-grid-column: 1;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(2) {
  -ms-grid-column: 3;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(3) {
  -ms-grid-column: 1;
  -ms-grid-row: 3;
}

.c0 > *:nth-child(4) {
  -ms-grid-column: 3;
  -ms-grid-row: 3;
}

.c0 > *:nth-child(5) {
  -ms-grid-column: 1;
  -ms-grid-row: 5;
}

.c0 > *:nth-child(6) {
  -ms-grid-column: 3;
  -ms-grid-row: 5;
}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

<div
    className="c0"
    display="inline-grid"
    rows={
      Array [
        "repeat(3, 1fr)",
        null,
        null,
        null,
        null,
      ]
    }
  >
    <div
      className=""
    >
      Products
    </div>
    <div
      className=""
    >
      Company
    </div>
    <div
      className=""
    >
      Sitemap
    </div>
    <div
      className=""
    >
      Contacts
    </div>
    <div
      className=""
    >
      Disclaimer
    </div>
    <div
      className=""
    >
      Privacy
    </div>
  </div>,
  .c0 {
  display: grid;
  display: -ms-grid;
  grid-template-columns: repeat(12,1fr);
  grid-gap: 3rem;
  -ms-grid-columns: 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr;
}

-ms-grid-rows:1fr .c0 > *:nth-child(1) {
  -ms-grid-column: 1;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(2) {
  -ms-grid-column: 3;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(3) {
  -ms-grid-column: 5;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(4) {
  -ms-grid-column: 7;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(5) {
  -ms-grid-column: 9;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(6) {
  -ms-grid-column: 11;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(7) {
  -ms-grid-column: 13;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(8) {
  -ms-grid-column: 15;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(9) {
  -ms-grid-column: 17;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(10) {
  -ms-grid-column: 19;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(11) {
  -ms-grid-column: 21;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(12) {
  -ms-grid-column: 23;
  -ms-grid-row: 1;
}

.c2 {
  display: none;
}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {
  .c0 {
    grid-template-columns: repeat(12,1fr);
  }
}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {
  .c0 {
    -ms-grid-columns: 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr;
  }

  -ms-grid-rows:1fr .c0 > *:nth-child(1) {
    -ms-grid-column: 1;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(2) {
    -ms-grid-column: 3;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(3) {
    -ms-grid-column: 5;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(4) {
    -ms-grid-column: 7;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(5) {
    -ms-grid-column: 9;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(6) {
    -ms-grid-column: 11;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(7) {
    -ms-grid-column: 13;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(8) {
    -ms-grid-column: 15;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(9) {
    -ms-grid-column: 17;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(10) {
    -ms-grid-column: 19;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(11) {
    -ms-grid-column: 21;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(12) {
    -ms-grid-column: 23;
    -ms-grid-row: 1;
  }
}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {
  .c1 {
    grid-column: 1/8;
  }
}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {
  .c1 {
    -ms-grid-column: 1 !important;
    -ms-grid-column-span: 14 !important;
  }
}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:48em) {
  .c2 {
    display: block;
  }
}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {
  .c2 {
    grid-column: 9/13;
  }
}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {
  .c2 {
    -ms-grid-column: 17 !important;
    -ms-grid-column-span: 8 !important;
  }
}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

<div
    className="c0"
    rows={
      Array [
        "repeat(1, 1fr)",
        null,
        "repeat(1, 1fr)",
        null,
        null,
      ]
    }
  >
    <div
      className="c1"
    >
      AAAAAAA
    </div>
    <div
      className="c2"
      display={
        Object {
          "_": "none",
          "sm": "block",
        }
      }
    >
      BBBB
    </div>
  </div>,
  .c0 {
  margin-bottom: 3rem;
  display: grid;
  display: -ms-grid;
  grid-template-columns: repeat(1,1fr);
  grid-gap: 3rem;
  -ms-grid-columns: 1fr;
}

-ms-grid-rows:1fr 3rem 1fr .c0 > *:nth-child(1) {
  -ms-grid-column: 1;
  -ms-grid-row: 1;
}

.c0 > *:nth-child(2) {
  -ms-grid-column: 1;
  -ms-grid-row: 3;
}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:48em) {
  .c0 {
    margin-bottom: 8rem;
  }
}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {
  .c0 {
    grid-template-columns: repeat(12,1fr);
  }
}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {
  .c0 {
    -ms-grid-columns: 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr 3rem 1fr;
  }

  -ms-grid-rows:1fr .c0 > *:nth-child(1) {
    -ms-grid-column: 1;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(2) {
    -ms-grid-column: 3;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(3) {
    -ms-grid-column: 5;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(4) {
    -ms-grid-column: 7;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(5) {
    -ms-grid-column: 9;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(6) {
    -ms-grid-column: 11;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(7) {
    -ms-grid-column: 13;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(8) {
    -ms-grid-column: 15;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(9) {
    -ms-grid-column: 17;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(10) {
    -ms-grid-column: 19;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(11) {
    -ms-grid-column: 21;
    -ms-grid-row: 1;
  }

  .c0 > *:nth-child(12) {
    -ms-grid-column: 23;
    -ms-grid-row: 1;
  }
}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {
  .c1 {
    grid-column: 1/6;
  }
}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {
  .c1 {
    -ms-grid-column: 1 !important;
    -ms-grid-column-span: 10 !important;
  }
}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {
  .c2 {
    grid-column: 6/13;
  }
}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:85em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {

}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

@media screen and (min-width:30em) {

}

@media screen and (min-width:48em) {
  .c2 {
    -ms-grid-column: 11 !important;
    -ms-grid-column-span: 14 !important;
  }
}

@media screen and (min-width:64em) {

}

@media screen and (min-width:78em) {

}

<div
    className="c0"
    rows={
      Array [
        "repeat(2, 1fr)",
        null,
        "repeat(1, 1fr)",
        null,
        null,
      ]
    }
  >
    <div
      className="c1"
    >
      AAAAAA
    </div>
    <div
      className="c2"
    >
      BBBBBB
    </div>
  </div>,
]
`;
